gocse/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/joho/godotenv"
+	"github.com/mongodb-university/csfle-guides/gocse/csfle"
+	"github.com/mongodb-university/csfle-guides/gocse/kms"
+	"github.com/mongodb-university/csfle-guides/gocse/patient"
+	"github.com/mongodb-university/csfle-guides/gocse/schema"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func localMasterKey() []byte {
+	if _, err := os.Stat("master-key.txt"); err == nil {
+		return getMasterKey()
+	}
+	key := make([]byte, 96)
+	if _, err := rand.Read(key); err != nil {
+		log.Fatalf("Unable to create a random 96 byte data key: %v", err)
```

Comment:
This applies here and everywhere `log.Fatalf` is used: it'd be preferred to return errors instead. Having `Fatalf` calls in random places makes it difficult to follow program flow. The only thing that should be intentionally exiting the program is `main()`.

That being said, these are technically setup functions, so any errors here are entirely user error and exiting could be warranted. If that's the case, use `panic` instead. Panics allow deferred functions to execute so you can clean up any open resources, while `Fatal` causes the program to exit immediately and skips deferred calls.

Reply:
Thanks. I've updated to `panic` once there is a client, and keeping it at `fatal` in the setup.